apps/electron-app/src/types/metadata.ts

```
@@ -0,0 +1,310 @@
+   * Safely extracts specific metadata type
+   */
+  static extractMetadata<T extends SuggestionMetadata>(
+    metadata: unknown,
+    validator: (data: unknown) => data is T,
+  ): T | null {
+    if (validator(metadata)) {
+      return metadata;
+    }
+    return null;
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)